python/tests/test_dataset.py

```diff
@@ -128,8 +128,8 @@ def test_del_item_missing():
 def test_coord_setitem():
     var = sc.Variable(dims=['x'], values=np.arange(4))
     d = sc.Dataset({'a': var}, coords={'x': var})
-    with pytest.raises(RuntimeError):
-        d['x', 2:3].coords['y'] = sc.Variable(1.0)
+    d['x', 2:3].coords['y'] = sc.Variable(1.0)  # no effect
+    assert 'y' not in d.coords
```

Comment:
Why did this change? Does this have to be silent for some use cases?

Reply:
It is silent for significantly simplifying the implemention: If we wanted to check this the meta data dicts would need to store references to their parent dataset or data array. But even that is not enough, since once you do something like
```python
ds['b'].attrs['attr']
a = ds['a']
a.coords['attr'] = var
```
things get weird: `a` has its own coords, so it should be allowed to add the coord there (since it does NOT add the coord to `ds.coords`)... but we still have the parent pointer so this may be prevented? And then we start turning in circles.

Reply:
Oh, I thought this was related to partial slices since the test uses `2:3`.

Reply:
Ah, wait, when I wrote the answer I indeed had the wrong example in mind (I was thinking of the case of silently adding a name-clashing attr or coord).

Reply:
The reason is partially similar though. Consider
```python
sliced = d['x', 2:3]
sliced.coords['y'] = sc.Variable(1.0)
```
With the new buffer-sharing mechanism, `sliced` pretty much behaves like a regular dataset. The second line simply adds a new coord to the slice. This does not affect `d`, since `sliced` has its own dict of coords (containing slices of `d`'s coords).

In Python we cannot distinguish this from `d['x', 2:3].coords['y'] = sc.Variable(1.0)` and the slice (and the new `y`) coord simply go out of scope silently.

Reply:
Ah, right. Somehow in my mind, it looked like this was changing and existing coordinate and not adding a new one. But what you said makes perfect sense.

Reply:
Good point though, would anyone expect that an existing coord would be changed? Try this example:

```python
import scipp as sc
var = sc.array(dims=['x'], values=[1.1,2.2,3.3])
a = sc.DataArray(data=var, coords={'x':var.copy()})
a['x', 1:2].coords['x'] = 1.8 * sc.units.m
a
```

Reply:
And consider that this *does* change `a.coords['x']`:
```python
a['x', 1:2].coords['x'] += 1.8 * sc.units.one
```

Reply:
This is pretty strange. Is this intentional or an accident from how `set()` works?

Reply:
And finally, the canonical(?) way to actually change a slice of the coord:
```python
a.coords['x']['x', 1:2] = 1.8 * sc.units.one
```

Reply:
> This is pretty strange
I think it is kind of "natural" in how Python works?
- `a['x', 1:2].coords['x'] = ...` calls `__setitem__` on the coords dict, i.e., adds (or replaces) a coord
- `a['x', 1:2].coords['x'] += ...` calls `__iadd__` (changing the coord in `a`), then `__setitem__` on the dict.
- `a.coords['x']['x', 1:2] = ...` calls `__setitem__` on the *coord*, setting a slice.

Reply:
Yes, kind of natural once one looks at the details but confusing nonetheless. I don't think we should change it because that could ultimately be much worse.